File: leetcode/206-reverse-linked-list/main.go
/*
Problem: Reverse Linked List
Link:	https://leetcode.com/problems/reverse-linked-list/description/
*/
package main

type ListNode struct {
	Val int
	Next *ListNode
}

func reverseList(head *ListNode) *ListNode {
  var prev *ListNode

  for head != nil {
    next := head.Next
    head.Next = prev
    prev = head
    head = next
  }

  return prev
}

File: leetcode/206-reverse-linked-list/Main.kt
package main

class Solution {
    fun reverseList(head: ListNode?): ListNode? {
        var currentNode: ListNode? = head
        var prevNode: ListNode? = null

        while (currentNode != null) {
            val nextNode = currentNode.next
            currentNode.next = prevNode
            prevNode = currentNode
            currentNode = nextNode
        }

        return prevNode
    }
}
